onebusaway-android/src/main/java/org/onebusaway/android/ui/HomeActivity.java

```diff
@@ -1579,14 +1582,54 @@ public void onClick(View arg0) {
         }
     }
 
-    private void checkDisplayZoomControls() {
+    private void checkDisplayZoomControls(boolean leftHandMode) {
         boolean displayZoom = Application.getPrefs().getBoolean(
                 getString(R.string.preference_key_show_zoom_controls), false);
 
-        mMapFragment.displayZoomControl(displayZoom);
+        LinearLayout zoomButtonsLayout = (LinearLayout) findViewById(R.id.zoom_buttons_layout);
+
+        if(displayZoom) {
+            zoomButtonsLayout.setVisibility(LinearLayout.VISIBLE);
+        } else {
+            zoomButtonsLayout.setVisibility(LinearLayout.GONE);
+        }
+
+        RelativeLayout.LayoutParams layoutParams = (RelativeLayout.LayoutParams) zoomButtonsLayout
+                .getLayoutParams();
+
+
+        int marginDp = (int) getResources().getDimension(R.dimen.zoom_btn_layout_margin);
+
+        if(leftHandMode) {
+
+            layoutParams.addRule(RelativeLayout.ALIGN_PARENT_LEFT, RelativeLayout.TRUE);
+            layoutParams.addRule(RelativeLayout.ALIGN_PARENT_START, RelativeLayout.TRUE);
+
+            if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.JELLY_BEAN_MR1) {
+                layoutParams.removeRule(RelativeLayout.ALIGN_PARENT_RIGHT);
+                layoutParams.removeRule(RelativeLayout.ALIGN_PARENT_END);
+            }
+
+            layoutParams.setMarginStart(marginDp);
+            layoutParams.setMargins(marginDp, 0, 0, 0);
+
+        } else /*not left hand mode*/ {
```

Comment:
You can eliminate the comment here, I think it's clear enough given the boolean name